internal/webserver/nodes.go

```diff
@@ -32,82 +35,112 @@ func (n kubeFilter) registerNode(router *mux.Router) {
 	})
 }
 
-func (n kubeFilter) nodeListHandler(_ http.ResponseWriter, request *http.Request) {
-	username, _, _ := req.NewHTTP(request, n.usernameClaimField).GetUserAndGroups()
-	selector := n.nodeSelector(request)
-	n.handleRequest(request, username, selector)
+func (n kubeFilter) getNodeSelector(nl *corev1.NodeList, selectors []map[string]string) (*labels.Requirement, error) {
+	var names []string
+	for _, node := range nl.Items {
+		for _, selector := range selectors {
+			matches := 0
+			for k := range selector {
+				if selector[k] == node.GetLabels()[k] {
+					matches++
+				}
+			}
+			if matches == len(selector) {
+				names = append(names, node.GetName())
+			}
+		}
+	}
+	if len(names) > 0 {
+		return labels.NewRequirement("kubernetes.io/hostname", selection.In, names)
```

Comment:
I guess we're doing this due to listing, right?

My idea was to construct essentially the LabelSelector adding barely the `map[string]string` as _Requirement_ but that wouldn't work.

If it is, LGTM.

Reply:
yes, because a user can own multiple tenants and each of them may have a different `nodeSelector`.

Reply:
Great, going to merge this, although squashing the commit to change the message.